clouddriver-huaweicloud/src/main/java/com/netflix/spinnaker/clouddriver/huaweicloud/security/HuaweiCloudCredentialsInitializer.java

```diff
@@ -0,0 +1,89 @@
+ * limitations under the License.
+ */
+
+package com.netflix.spinnaker.clouddriver.huaweicloud.security;
+
+import com.netflix.spinnaker.clouddriver.huaweicloud.config.HuaweiCloudConfigurationProperties;
+import com.netflix.spinnaker.clouddriver.huaweicloud.config.HuaweiCloudConfigurationProperties.ManagedAccount;
+import com.netflix.spinnaker.clouddriver.security.AccountCredentialsRepository;
+import com.netflix.spinnaker.clouddriver.security.CredentialsInitializerSynchronizable;
+import com.netflix.spinnaker.clouddriver.security.ProviderUtils;
+import java.util.List;
+import java.util.stream.Collectors;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.config.ConfigurableBeanFactory;
+import org.springframework.boot.context.properties.ConfigurationProperties;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Configuration;
+import org.springframework.context.annotation.Scope;
+
+@Configuration
+public class HuaweiCloudCredentialsInitializer implements CredentialsInitializerSynchronizable {
+
+  private final Logger log = LoggerFactory.getLogger(HuaweiCloudCredentialsInitializer.class);
+
+  @Scope(ConfigurableBeanFactory.SCOPE_PROTOTYPE)
+  @Bean
+  @ConfigurationProperties("huaweicloud")
+  HuaweiCloudConfigurationProperties huaweiCloudConfigurationProperties() {
+    return new HuaweiCloudConfigurationProperties();
+  }
+
+  @Bean
+  List<HuaweiCloudNamedAccountCredentials> synchronizeHuaweiCloudNamedAccountCredentials(
+      HuaweiCloudConfigurationProperties huaweiCloudConfigurationProperties,
+      AccountCredentialsRepository accountCredentialsRepository) {
+
+    List result =
+        ProviderUtils.calculateAccountDeltas(
+            accountCredentialsRepository,
+            HuaweiCloudNamedAccountCredentials.class,
+            huaweiCloudConfigurationProperties.getAccounts());
+
+    List<ManagedAccount> accountsToAdd = (List<ManagedAccount>) result.get(0);
+    accountsToAdd.forEach(
+        managedAccount -> {
+          try {
+            HuaweiCloudNamedAccountCredentials account =
+                new HuaweiCloudNamedAccountCredentials(
+                    managedAccount.getName(),
+                    managedAccount.getEnvironment(),
+                    managedAccount.getAccountType(),
+                    managedAccount.getAuthUrl(),
+                    managedAccount.getUsername(),
+                    managedAccount.getPassword(),
+                    managedAccount.getProjectName(),
+                    managedAccount.getDomainName(),
+                    managedAccount.getInsecure(),
+                    managedAccount.getRegions());
+
+            accountCredentialsRepository.save(managedAccount.getName(), account);
+          } catch (Exception e) {
+            log.error(
+                "Could not load account %s for huaweicloud, error=%s", managedAccount.getName(), e);
```

Comment:
nit : `log.error("Could not load account: {} for huaweicloud", managedAccount.getName(), e)`

Reply:
Done